### PR TITLE
[WIP] adding ansible tab -- initial pr

### PR DESCRIPTION
**Pull request by zikalino**
This PR adds additional tab with Ansible playbooks equivalent to PowerShell tab.
Users can copy & paste playbooks and use them in production environment.

### CODE REVIEW

**Comment by davidebbo** [comment]
Would be good for @bala16 to take a look, but he's out till end of next week. Taking a glance at the diff, it looks quite isolated, so I don't think there is much concern about getting it in.

Just one potential issue: I think we only have up to TypeScript 2.5 in App Service, so if we change it to 2.8, build might break. Of course we probably should get the latest TypeScript in App Service, but unless it's really needed, it may be simpler to just stick with 2.5. @ahmelsayed may have more to say about this.

**Comment by zikalino** [comment]
@davidebbo i will change version back to 2.5. i have just updated it as my VS was bothering me to do so....
i have also sent you an e-mail with screenshot of this solution.
i will do some more tests and still improve a few smaller things in a meanwhile...

File: ng/scriptGenerators/ansible/AnsibleScriptGenerator.ts [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+                    break;
+                }
+                case CmdType.New: {
+                    if (cmdActionPair.isSetAction) {
+                        currentScript += '    - name: SET ' + this.resolver.getActionName() + '\n';
+                        currentScript += '      azure_rm_resource:\n';
+                        currentScript += this.yamlFromResourceId("        ");
+                    }
+                    else {
+                        let newName: string = "New" + this.resolver.getResourceName();
```
Comment (bala16) [inline ng/scriptGenerators/ansible/AnsibleScriptGenerator.ts]: not used anywhere.

File: ng/scriptGenerators/ansible/AnsibleScriptGenerator.ts [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+                        currentScript += '    - name: CREATE ' + this.resolver.getActionName() + '\n';
+                        currentScript += '      azure_rm_resource:\n';
+                        // XXX - resolve this
```
Comment (bala16) [inline ng/scriptGenerators/ansible/AnsibleScriptGenerator.ts]: is this is a todo? can you open issues to track this instead of commenting here? also at Line 84

File: ng/scriptGenerators/ansible/AnsibleScriptGenerator.ts [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+                    yaml += prefix + key + ": " + o[key] + "\n";
+                }
+            }
+
+            return yaml;
+        }
+
+        private yamlFromResourceId(prefix: string): string {
+            let yaml: string = "";
+            let cmdParameters = this.resolver.getParameters();
```
Comment (bala16) [inline ng/scriptGenerators/ansible/AnsibleScriptGenerator.ts]: can be const

**Comment by zikalino** [comment]
@bala16 fixed a few things + did more testing. I think I managed to test all the corner cases. could we stage it now?

**Comment by bala16** [comment]
@zikalino Thanks. can you also review the comments i made earlier. let me know if you can't get to it , i can make the changes and merge.

File: ng/scriptGenerators/ansible/AnsibleScriptGenerator.ts (hunk 1 above)
Comment (zikalino) [reply ng/scriptGenerators/ansible/AnsibleScriptGenerator.ts]: removed now, sorry missed it....

File: ng/scriptGenerators/ansible/AnsibleScriptGenerator.ts (hunk 2 above)
Comment (zikalino) [reply ng/scriptGenerators/ansible/AnsibleScriptGenerator.ts]: it's now resolved properly, so no need for todo.

File: ng/scriptGenerators/ansible/AnsibleScriptGenerator.ts (hunk 3 above)
Comment (zikalino) [reply ng/scriptGenerators/ansible/AnsibleScriptGenerator.ts]: I made replacements wherever necessary

**Comment by zikalino** [comment]
@bala16 now I think everything is corrected.

**Comment by bala16** [comment]
available on the staging site now - https://resources-staging.azure.com/
